File: szulogyerek.cpp
#include<iostream>

using namespace std;


class Szulo{
	public: void uzenet(){
	cout<<"I'm the parent class"<<endl;
	}
};

void fgv ( Szulo& parent) {
	parent.uzenet();
}

class Gyerek: public Szulo{
	public: void uzenet(){
	cout<<"I'm the child class"<<endl;
	}
};

int main( int argc, char **argv){

	Szulo* szulo=new Gyerek();
	szulo->uzenet();
	
	Gyerek gyerek;
	fgv(gyerek);	
}

File: Szulogyerek.java
class Szulo{
	public void uzenet(){
	System.out.println("I'm the parent class");}
}

class Gyerek extends Szulo{
	public void uzenet2(){
	System.out.println("I'm the child class");}
}

class Szulogyerek{
	public static void main (String[] args){

	Szulo szulo=new Gyerek();
	szulo.uzenet();
	//szulo.uzenet2();

	}
}
